### PR TITLE
COMPAT: Categorical Subclassing

### PR DESCRIPTION
**Pull request by sinhrks**
- [x] related to #8640
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`

It allows to subclass `Categorical` related to #8640.  Shoud it use `_simple_new` or `_constructor`?

### CODE REVIEW

**Comment by shoyer** [comment]
The changes in this PR are fine, but I'm not sure it's a good idea to subclassing Categorical in the API, though I would guess nothing major is likely to change before v2.0. `Categorical` is diving pretty far into pandas internals.

**Comment by sinhrks** [comment]
It may bettr to discuss in #8640, I' m thinking to the following stricture:

- CategoricalMixin ( move internal methods related to `_categories` and `_codes`)
   - Categorical (no API change)
   - String (automatically handle `_categories`, `_codes` and `.str` methods)

I don't think we have to wait for 2.0 as long as it does't break existing user's code, but it also should be discussed separatelly (in ML).

File: pandas/core/categorical.py [hunk 1]
```diff
@@ -330,9 +330,9 @@ def __init__(self, values, categories=None, ordered=False,
 
     def copy(self):
         """ Copy constructor. """
-        return Categorical(values=self._codes.copy(),
```
Comment (jreback) [inline pandas/core/categorical.py]: maybe make a ``_constructor`` as that is the patter we are using for other classes

**Comment by sinhrks** [comment]
Updated to use ``_constructor`` and added small number of tests.

**Comment by jreback** [comment]
ty!

**Pull request closed without merging**